lua/nordic/tests/options.lua

```diff
@@ -1,55 +1,168 @@
+load({ after_palette = function(_palette) end })
+assert_eq(highlight('Normal').bg, '#242933', 'after_palette: reloading should revert the palette to its original state')
+
+-- on_highlight
+assert_eq(highlight('Normal').bg, '#242933', 'on_highlight: `Normal` should be `#242933` by default')
+load({ on_highlight = function(highlights, _palette) highlights['Normal'].bg = '#FFFFFF' end })
+assert_eq(highlight('Normal').bg, '#FFFFFF', 'on_highlight: changing a highlight should actually change the highlight')
+load({ on_highlight = function(_highlights, _palette) end })
+assert_eq(highlight('Normal').bg, '#242933',
+  'on_highlight: reloading should revert all highlights to their original state')
+
+-- bold_keywords
+assert_eq(highlight('Keyword').bold, nil, 'bold_keywords: highlight `Keyword` should not be bold by default')
+load({ bold_keywords = true })
+assert_eq(highlight('Keyword').bold, true, 'bold_keywords: highlight `Keyword` should be bold if `bold_keywords` is true')
+load({ bold_keywords = false })
+
+-- italic_comments
+assert_eq(highlight('Comment').italic, true, 'italic_comments: highlight `Comments` should be italic by default')
+load({ italic_comments = false })
+assert_eq(highlight('Comment').italic, nil,
+  'italic_comments: highlight `Comments` should not be italic if `italic_comments` is false')
+load({ italic_comments = true })
+
+-- transparent
+-- bg
+assert_eq(highlight('Normal').bg ~= nil, true, 'transparent: highlight `Normal.bg` should not be `nil` by default')
+load({ transparent = { bg = true } })
+assert_eq(highlight('Normal').bg, nil, 'transparent: highlight `Normal.bg` should be `nil` if `transparent.bg` is true')
+load({ transparent = { bg = false } })
+-- float
+assert_eq(highlight('NormalFloat').bg ~= nil, true,
+  'transparent: highlight `NormalFloat.bg` should not be `nil` by default')
+load({ transparent = { float = true } })
+assert_eq(highlight('NormalFloat').bg, nil,
+  'transparent: highlight `NormalFloat.bg` should be `nil` if `transparent.float` is true')
+load({ transparent = { float = false } })
+
+-- bright_border
+assert_eq(highlight('WinSeparator').fg, '#191D24',
+  'bright_border: all highlights that use `border_fg` should be `#191D24` by default')
+load({ bright_border = true })
+-- note: this could fail if the wrong white0 variant is used
+assert_eq(highlight('WinSeparator').fg, '#C0C8D8',
+  'bright_border: all highlights that use `border_fg` should be `#C0C8D8` if `bright_border` is true')
+load({ bright_border = false })
+
+-- reduced_blue
+assert_eq(highlight('Normal').fg, '#C0C8D8',
+  'reduced_blue: all highlights that use `white0` should be `#C0C8D8` by default')
+load({ reduced_blue = false })
+assert_eq(highlight('Normal').fg, '#BBC3D4',
+  'reduced_blue: all highlights that use `white0` should be `#BBC3D4` if `reduced_blue` is false')
+load({ reduced_blue = true })
+
+-- swap_backgrounds
+-- note: this could fail if any transparent settings are set
```

Comment:
Yes this definitely fails.  But that's fine, we control the config in the tests.  Maybe just change the comment to say it will fail.

Just on that note - we need to think about having theme colors hardcoded here.  If any of the mentioned hl groups change or any of the colors we use change the tests will fail.  Basically, our theme colors are now being "described" in two seperate places and we need to maintain the two seperate places.

Reply:
For a start we can load in all of the palette colors and use them as reference, not hard coded values (obviously where we do not overwrite the colors)

Reply:
I fixed the comments in [docs(comments): improve wording;](https://github.com/AlexvZyl/nordic.nvim/pull/150/commits/e5c5098d98404376c4c75802fb7d2defe33a01bc).
I am not sure why I didn't just import the palette to start with, but I will fix that in a few minutes.

Note: All hex codes in `colors/nordic.lua` must be uppercase to work with `get_highlight`, we can change that, but it's more code than I think it's worth. For now, I will just add a comment.

Reply:
see: [fix(tests): replace hard-coded colors;](https://github.com/AlexvZyl/nordic.nvim/pull/150/commits/c9b23864c39a6605cc60cdf5e6136bbd28d78e93)